server/lib/ga-collector.js

```diff
@@ -103,6 +106,30 @@ GACollector.prototype = {
         });
       }
     });
+  },
+  /**
+   * Calculate the queue time of the event.
+   * See https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters?hl=en#qt
+   *
+   * @param {Number} startTime
+   *  Time when metrics object was create on the front-end
+   * @param {Number} offset
+   *  Offset of the event since metrics start
+   *
+   * @returns {Number}
+   *  The time delta (in milliseconds) between when the hit being reported occurred and the time the hit was sent.
+   *
+   * @private
+   */
+  _calculateQueueTime: function(startTime, offset) {
+    if (! startTime || ! offset) {
+      return 0;
+    }
+
+    var dispatchTime = Date.now();
+    var offsetTime = startTime + offset;
```

Comment:
> clock skew between the server and client

1000 times this, clients routinely have clock skew on the order of minutes or even hours